rustls/src/server/builder.rs

```diff
@@ -15,6 +16,19 @@ use alloc::vec::Vec;
 use core::marker::PhantomData;
 
 impl ConfigBuilder<ServerConfig, WantsVerifier> {
+    /// Create builder to build up the `webpki` client certificate verifier configuration.
+    /// Client certificate authentication will be offered by the server, and client certificates
+    /// will be verified using the trust anchors found in the provided `roots`. If you
+    /// wish to disable client authentication use [ConfigBuilder<ServerConfig, WantsVerifier>::with_no_client_auth()] instead.
+    ///
+    /// For more information, see the [`webpki::ClientCertVerifierBuilder`] documentation.
+    pub fn build_webpki_verifier(
```

Comment:
This seems like a pretty idiosyncratic API. I think I'd prefer mirroring the `ServerConfig` and `ClientConfig` `builder()` and `builder_with_provider()` constructors.

Reply:
Yes, it's not ideal. I'm trying to avoid a situation where one provider is used for the config, and another is used for the verifier. (As a concrete example, someone asks for a FIPS provider for the config, but neglects to do that for the verifier. The result would be something that isn't FIPS approved.)

Reply:
I guess the `ClientCertVerifierBuilder::new` could take a `ConfigBuilder<...>`?

Reply:
Hmm, by reference? Seems like it could be okay... Still pretty tangled but probably better than this.

Reply:
> As a concrete example, someone asks for a FIPS provider for the config, but neglects to do that for the verifier. The result would be something that isn't FIPS approved.

Having thought about this a bit more, I think this is insufficient. It makes less likely, but still possible, to have a FIPS-approved `CryptoProvider` under the `{Client,Server}Config`, but with a different one under the verifier. So I'm going to put this back how it was, and add a `CryptoProvider` parameter to the previous `WebPkiClientVerifier::builder` constructor.

I think I have some better ideas about how to detect that case, merged in with the need to have a "FIPS service indicator": basically that each `{Client,Server}Connection` can say "yes, I'm in FIPS mode" with a good degree of accuracy.

Reply:
So this commit is dropped, and instead we have `WebPkiServerVerifier::builder_with_provider`